contracts/DCAHubSwapper/DCAHubSwapperSwapHandler.sol

```diff
@@ -96,12 +96,22 @@ abstract contract DCAHubSwapperSwapHandler is DeadlineValidation, AccessControl,
   }
 
   /// @inheritdoc IDCAHubSwapperSwapHandler
-  function swapWithDexes(SwapWithDexesParams calldata _parameters) external payable returns (IDCAHub.SwapInfo memory) {
+  function swapWithDexes(SwapWithDexesParams calldata _parameters)
+    external
+    payable
+    onlyRole(SWAP_EXECUTION_ROLE)
+    returns (IDCAHub.SwapInfo memory)
+  {
     return _swapWithDexes(_parameters, false);
   }
 
   /// @inheritdoc IDCAHubSwapperSwapHandler
-  function swapWithDexesForMean(SwapWithDexesParams calldata _parameters) external payable returns (IDCAHub.SwapInfo memory) {
+  function swapWithDexesForMean(SwapWithDexesParams calldata _parameters)
+    external
+    payable
+    onlyRole(SWAP_EXECUTION_ROLE)
+    returns (IDCAHub.SwapInfo memory)
+  {
```

Comment:
If we are doing this permissioned, is it worth having the "non-for-mean"? 👀

Reply:
I think so. We might allow non-mean users to use this in the future, and it's already coded haha

Reply:
Oh. You mean giving auth to someone that is not mean? Like an external MM for example?

Reply:
If that's what you mean Sam, then I guess it makes sense